java/ql/lib/semmle/code/java/dataflow/internal/ModelExclusions.qll

```diff
@@ -59,12 +59,26 @@ private predicate isInternal(CompilationUnit cu) {
   cu.getPackage().getName().matches("%internal%")
 }
 
+/** A method relating to lambda flow. */
+private class LambdaFlowMethod extends Method {
+  LambdaFlowMethod() {
+    this.hasQualifiedName("java.lang", "Runnable", "run") or
+    this.hasQualifiedName("java.util", "Comparator", "comparing") or
+    this.hasQualifiedName("java.util.function", "BiConsumer", "accept") or
+    this.hasQualifiedName("java.util.function", "BiFunction", "apply") or
+    this.hasQualifiedName("java.util.function", "Consumer", "accept") or
+    this.hasQualifiedName("java.util.function", "Function", "apply") or
+    this.hasQualifiedName("java.util.function", "Supplier", "get")
+  }
+}
+
 /** Holds if the given callable is not worth modeling. */
 predicate isUninterestingForModels(Callable c) {
   isInTestFile(c.getCompilationUnit().getFile()) or
   isInternal(c.getCompilationUnit()) or
   c instanceof MainMethod or
   c instanceof ToStringMethod or
+  c instanceof LambdaFlowMethod or
```

Comment:
Would it make sense to use [`FunctionalInterface`](https://github.com/github/codeql/blob/a97510a2fc3659567dfe7b280d3e782fb3bb4b67/java/ql/lib/semmle/code/java/Type.qll#L1033) instead?

Reply:
Hm, nvm, some of those aren't functional interfaces.

Reply:
Hmm, I might be able to use that to replace some of the hard-coded method names. I'll have a play around with it.

Reply:
It gave almost 7000 results! I think it's best to just hard-code a few for now and if we find we want to do more in future then we can come up with a clever way to characterise them.

Reply:
Makes sense 👍